etcdserver/auth/auth.go

```diff
@@ -68,6 +68,12 @@ var guestRole = Role{
 	},
 }
 
+var (
+	// These fields exist to simplify testing.
```

Comment:
the comment is confusing. I think we add these indirection fields for switchable. So the test can switch them to whatever tests wants. It will not simplify testing at all.